--- config.m4.org	2014-01-23 09:31:20.148938076 +0100
+++ config.m4	2014-01-23 09:37:37.628332947 +0100
@@ -21,6 +21,7 @@
 proxy_fdpass_objs="mod_proxy_fdpass.lo"
 proxy_ajp_objs="mod_proxy_ajp.lo ajp_header.lo ajp_link.lo ajp_msg.lo ajp_utils.lo"
 proxy_balancer_objs="mod_proxy_balancer.lo"
+proxy_cluster_objs="mod_proxy_cluster.lo"
 
 case "$host" in
   *os2*)
@@ -34,6 +35,7 @@
     proxy_fdpass_objs="$proxy_fdpass_objs mod_proxy.la"
     proxy_ajp_objs="$proxy_ajp_objs mod_proxy.la"
     proxy_balancer_objs="$proxy_balancer_objs mod_proxy.la"
+    proxy_cluster_objs="$proxy_cluster_objs mod_proxy.la"
     ;;
 esac
 
@@ -54,7 +56,7 @@
 ],proxy)
 APACHE_MODULE(proxy_ajp, Apache proxy AJP module.  Requires and is enabled by --enable-proxy., $proxy_ajp_objs, , $proxy_mods_enable,, proxy)
 APACHE_MODULE(proxy_balancer, Apache proxy BALANCER module.  Requires and is enabled by --enable-proxy., $proxy_balancer_objs, , $proxy_mods_enable,, proxy)
-
+APACHE_MODULE(proxy_cluster, proxy CLUSTER module  Requires and is enabled by --enable-proxy., $proxy_cluster_objs, , $proxy_mods_enable,, proxy)
 APACHE_MODULE(proxy_express, mass reverse-proxy module. Requires --enable-proxy., , , $proxy_mods_enable,, proxy)
 
 APR_ADDTO(INCLUDES, [-I\$(top_srcdir)/$modpath_current])
